### ISSUE TITLE
Prettier sunset via RGB

### ISSUE THREAD

**Issue by RubenKelevra** [basnijholt/adaptive-lighting#131]
The current sunset/sunrise is pretty boring, even when using RGB mode.

I have three RGB lights and used manually scenes I switched through for a nice sunset/sunrise effect.

Is there a chance to have (somewhat randomly) generated sunrises and sunsets when RGB lights are available as an option?

I'm happy to supply some RGB values for scene steps with transition times!

**Comment by spencerthayer** [comment]
I am pretty sure @basnijholt isn't interested in issue #100 at all. And while I would prefer to be able to control transitions between time events, I would also enjoy being able to control RGB color transitions as well.

**Comment by RubenKelevra** [comment]
@spencerthayer I don't think there's a lack of interest. It's more a time and motivation constraint after being a bit disappointed (?) that his PR wasn't even looked at for month.

https://github.com/home-assistant/core/pull/40626#issuecomment-850399527

Anyway, those extremely high blue values in your graph are not possible with non-rgb lights anyway.

So it would be neat to combine both and have just three different types of lamps:

- fixed color temperate lamps (see #123) 
- rgb lamps which are used as ambient light
- color changing lights

With the lights marked as ambient lights, like LED strips etc. Adaptive Lighting could mimic either a non-linear color temperature change or a much more colorful one, depending on the user choices.

For Adaptive Lighting to calculate with your and my data, it needs to be converted from the currently time based info, to a degree of sun above/below the horizon.

Would you mind to extend your ticket with this info? 🤔

**Comment by basnijholt** [comment]
That seems like a good idea. I have already experimented with this in the past and wrote some code to linearly interpolate arbitrary colors spaces 😄. https://github.com/basnijholt/home-assistant-config/blob/master/pyscript/python_modules/wake_up_light.py 

It might take some time before I have time for this though. 

Also, if I had all the time in the world I would look at every issue and try to solve it, however this is not the case.

**Comment by spencerthayer** [comment]
> Anyway, those extremely high blue values in your graph are not possible with non-rgb lights anyway.

That's absolutely true. I suppose I am being a little short-sighted assuming everyone has RGB lighting throughout their house.

**Comment by RubenKelevra** [comment]
Well, I do use RGB lighting, but the RGB bulbs have are usually way too less light output for indirect lighting a room.

I use two 60 W LED (yes, real Watt) 6500 K light bulbs for indirect lighting my office, additionally to two Yeelight Bulb 1S for Ambient Light and some Aqara Zigbee Bulbs with 9W.

Only this gives enough flexibility without going bankrupt for getting the room brightly lit 😬

**Comment by RubenKelevra** [comment]
@spencerthayer

Well, I do use RGB lighting, but the RGB bulbs have are usually way too low light output for indirect lighting a room.

I use two 60 W LED (yes, real Watt) 6500 K light bulbs for indirect lighting my office, additionally to two Yeelight Bulb 1S for Ambient Light and some Aqara Zigbee Bulbs with 9W.

Only this gives enough flexibility without going bankrupt for getting the room brightly lit 😬

The two 60 W LED bulbs have together more than 4000 Lumen, while one Yeelight Bulb 1S outputs only 800 Lumen, but at 6500 K certainly a bit less IMHO.

The Aqara do have real 800 Lumen output at the end of the white color range.

**Comment by RubenKelevra** [comment]
@spencerthayer I'm currently in the process of doing version 2. This will be implemented in it.
